### ISSUE TITLE
UX - Editing notes is destructive

### ISSUE THREAD

**Issue by radumg** [DynamoDS/Dynamo#8221]
## Dynamo version

1.3.1

## Operating system

Windows 10.

## What did you do? 

Double-clicked a note
deleted all text 
clicked the modal window's close (X) button

## What did you expect to see?

The edit note window has a clear `Accept` button, so when closing the window by using the form's X, I expect the edits not to be saved, as the `Accept` button was never clicked.

I would like either
- the X does not save edits
- provide a cancel button

## What did you see instead?

the note now doesn't have any text and I have to manually undo.

# Extras
It would be great to solve this other little thing whilst at it, the modal window simply says `Set value...`. This is not descriptive at all, we should replace with something that clear tells the user what they are doing, perhaps something along the lines of `Edit note text` ?
![edit note](https://user-images.githubusercontent.com/11439624/31053274-a28e9e32-a691-11e7-9e55-2fc4dbc194a1.PNG)

**Comment by andydandy74** [comment]
See this discussion: #289

**Comment by radumg** [comment]
thanks @andydandy74 , definitely didn't know it had been raised before.

Perhaps this can server as further evidence this is indeed what users would expect - vast majority of modals on both Windows & Mac behave this way after all.

I would argue most users won't understand the value is bound precisely because it's a separate, modal window. Now, if editing the note directly was enabled, that would make it all clear - and everyone happy 😄

**Comment by andydandy74** [comment]
True on all accounts. Although I would think that even when editing notes directly in canvas, pressing Esc should cancel the edit.

**Comment by radumg** [comment]
Yes, altogether that would be much nicer & intuitive UX me thinks.
At this point, I'd settle for being able to vertically resize the edit note window 😭

**Comment by andydandy74** [comment]
@Racel Perhaps another item for your UX Love epic?

**Comment by Racel** [comment]
We will take a quick look at this. If it is a small fix, we will try and get it in for the next release. If not, we will make these considerations for the next generation UI.
